auditbeat/module/file_integrity/eventreader_linux.go

```diff
@@ -0,0 +1,61 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package file_integrity
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+func NewEventReader(c Config) (EventProducer, error) {
+	if c.Backend == BackendAuto || c.Backend == BackendFSNotify || c.Backend == "" {
+		// Auto and unset defaults to fsnotify
+		l := logp.NewLogger(fmt.Sprintf("%s__fsnotify", moduleName))
```

Comment:
I would join the name with dots. That's the same convention used by `Named()` (https://pkg.go.dev/go.uber.org/zap#Logger.Named). So if we had passed in a parent logger, and then called `parent.Named("fsnotify")` you would get the same result.

We probably should pass in the parent logger in case it includes some additional context like and `id`. The `BaseMetricSet` provides a logger so we should probably be using that as our starting point instead of calling `NewLogger`.